python36Packages.cffi: build is broken

**Describe the bug**

`python36Packages.cffi` doesn't build.

**To Reproduce**
Steps to reproduce the behavior:

Check out nixpkgs master (currently 1e678a98d56003bb73e7fb22bcae7eeff0bc699e) and run `nix build python36Packages.cffi`. Observe the following output:

```
$ nix build -f. python36Packages.cffi
builder for '/nix/store/jd8fm9kgavyi70wr5316hhvnl1rli681-python3.6-cffi-1.14.4.drv' failed with exit code 1; last 10 log lines:
  FAILED testing/cffi0/test_function.py::TestFunction::test_missing_function - ...
  FAILED testing/cffi0/test_function.py::TestFunction::test_wraps_from_stdlib
  FAILED testing/cffi0/test_function.py::TestFunction::test_stdcall_only_on_windows
  FAILED testing/cffi0/test_function.py::TestFunction::test_dir_on_dlopen_lib
  FAILED testing/cffi0/test_parsing.py::test_simple - OSError: ctypes.util.find...
  FAILED testing/cffi0/test_parsing.py::test_remove_comments - OSError: ctypes....
  FAILED testing/cffi0/test_parsing.py::test_remove_line_continuation_comments
  FAILED testing/cffi0/test_parsing.py::test_line_continuation_in_defines - OSE...
  FAILED testing/cffi0/test_unicode_literals.py::test_dlopen - OSError: ctypes....
  = 32 failed, 1906 passed, 98 skipped, 1 deselected, 4 xfailed, 207 warnings in 286.78s (0:04:46) =
[0 built (1 failed), 53 copied (119.0 MiB), 21.8 MiB DL]
error: build of '/nix/store/jd8fm9kgavyi70wr5316hhvnl1rli681-python3.6-cffi-1.14.4.drv' failed
```

**Expected behavior**
Package should build.

**Notify maintainers**
Pinging maintainers @domenkozar and @LnL7.

**Metadata**
The result of running `nix-shell -p nix-info --run "nix-info -m"`:

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.85, NixOS, 20.09.2382.ca119749d86 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(root): `"nixos-20.09.2382.ca119749d86"`
 - channels(fs): `"nixpkgs-21.03pre259722.a60a1340b7d"`
 - nixpkgs: `/home/kini/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: python36Packages.cffi
# a list of nixos modules affected by the problem
module:
```

#108243 was resolved by #108569, which is already merged into master. 349585e7787e84cf1fe745665a08cbf4d4f477d5 is an ancestor of 1e678a98d56003bb73e7fb22bcae7eeff0bc699e, the commit upon which I saw this error. Are you sure that this issue is already fixed?

In fact, I checked out the branch of PR #108569 and was still able to reproduce this build failure on that branch. (However, the Python 2.7 build of `cffi` succeeds.)

```
$ git fetch upstream refs/pull/108569/head &>/dev/null

$ git checkout FETCH_HEAD &>/dev/null

$ nix build -f. python36Packages.cffi
builder for '/nix/store/cxqsgk0rbkf3yv0p0jfvsmn6kbb3cgsb-python3.6-cffi-1.14.4.drv' failed with exit code 1; last 10 log lines:
  FAILED testing/cffi0/test_function.py::TestFunction::test_missing_function - ...
  FAILED testing/cffi0/test_function.py::TestFunction::test_wraps_from_stdlib
  FAILED testing/cffi0/test_function.py::TestFunction::test_stdcall_only_on_windows
  FAILED testing/cffi0/test_function.py::TestFunction::test_dir_on_dlopen_lib
  FAILED testing/cffi0/test_parsing.py::test_simple - OSError: ctypes.util.find...
  FAILED testing/cffi0/test_parsing.py::test_remove_comments - OSError: ctypes....
  FAILED testing/cffi0/test_parsing.py::test_remove_line_continuation_comments
  FAILED testing/cffi0/test_parsing.py::test_line_continuation_in_defines - OSE...
  FAILED testing/cffi0/test_unicode_literals.py::test_dlopen - OSError: ctypes....
  = 32 failed, 1906 passed, 98 skipped, 1 deselected, 4 xfailed, 207 warnings in 279.49s (0:04:39) =
[0 built (1 failed), 60 copied (302.1 MiB), 68.4 MiB DL]
error: build of '/nix/store/cxqsgk0rbkf3yv0p0jfvsmn6kbb3cgsb-python3.6-cffi-1.14.4.drv' failed

$ nix build -f. python27Packages.cffi
[46 built, 60 copied (338.2 MiB), 58.1 MiB DL]
```

I think this indicates that this issue is not a duplicate of #108243, which is after all specifically about Python 2. The fix in #108569 also only touches Python 2 -related files.

--- a/pkgs/development/interpreters/python/cpython/default.nix
+++ b/pkgs/development/interpreters/python/cpython/default.nix
@@ -216,7 +216,7 @@ in with passthru; stdenv.mkDerivation {
       else
         ./3.7/fix-finding-headers-when-cross-compiling.patch
     )
-  ] ++ optionals (isPy37 || isPy38) [
+  ] ++ optionals (isPy36 || isPy37 || isPy38) [
     # Backport a fix for ctypes.util.find_library.
     ./3.7/find_library.patch
   ];

